pkg/openshift/deploymentconfig.go

```diff
@@ -15,13 +22,92 @@ var _ model.Revertor = DeploymentConfigRevertor
 
 // DeploymentConfigLocator attempts to locate a DeploymentConfig kind based on Ref name
 func DeploymentConfigLocator(ctx model.SessionContext, ref *model.Ref) bool { //nolint[:hugeParam]
-	return false
+	deployment, err := getDeploymentConfig(ctx, ctx.Namespace, ref.Name)
+	if err != nil {
+		if errors.IsNotFound(err) { // Ref is not a Deployment type
+			return false
+		}
+		ctx.Log.Error(nil, "Could not get Deployment", "name", deployment.Name)
+		return false
+	}
+	return true
 }
 
+// DeploymentConfigMutator attempts to clone the located DeploymentConfig
 func DeploymentConfigMutator(ctx model.SessionContext, ref *model.Ref) error { //nolint[:hugeParam]
+	if len(ref.GetResourceStatus(DeploymentConfigKind)) > 0 {
+		return nil
+	}
+
+	deployment, err := getDeploymentConfig(ctx, ctx.Namespace, ref.Name)
+	if err != nil {
+		return err
+	}
+	ctx.Log.Info("Found Deployment", "name", ref.Name)
+
+	deploymentClone := cloneDeployment(deployment.DeepCopy())
+	err = ctx.Client.Create(ctx, deploymentClone)
+	if err != nil {
+		ctx.Log.Info("Failed to clone Deployment", "name", deploymentClone.Name)
+		ref.AddResourceStatus(model.ResourceStatus{Kind: DeploymentConfigKind, Name: deploymentClone.Name, Action: model.ActionFailed})
+		return err
+	}
+	ctx.Log.Info("Cloned Deployment", "name", deploymentClone.Name)
+	ref.AddResourceStatus(model.ResourceStatus{Kind: DeploymentConfigKind, Name: deploymentClone.Name, Action: model.ActionCreated})
 	return nil
 }
 
+// DeploymentConfigRevertor attempts to delete the cloned DeploymentConfig
 func DeploymentConfigRevertor(ctx model.SessionContext, ref *model.Ref) error { //nolint[:hugeParam]
+	statuses := ref.GetResourceStatus(DeploymentConfigKind)
+	for _, status := range statuses {
+		deployment := &appsv1.DeploymentConfig{
+			ObjectMeta: metav1.ObjectMeta{Name: status.Name, Namespace: ctx.Namespace},
+		}
+		ctx.Log.Info("Found DeploymentConfig", "name", ref.Name)
+		err := ctx.Client.Delete(ctx, deployment)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			ctx.Log.Info("Failed to delete DeploymentConfig", "name", deployment.Name)
+			ref.AddResourceStatus(model.ResourceStatus{Kind: DeploymentConfigKind, Name: deployment.Name, Action: model.ActionFailed})
+			return err
+		}
+		ref.RemoveResourceStatus(model.ResourceStatus{Kind: DeploymentConfigKind, Name: deployment.Name})
+	}
 	return nil
 }
+
+func cloneDeployment(deployment *appsv1.DeploymentConfig) *appsv1.DeploymentConfig {
```

Comment:
This duplication should be handled in #61

Reply:
Don't forget to align that with changes from #110